File: content/problems/0795-number-of-subarrays-with-bounded-maximum/solutions/difference-of-at-most-counts/solution.cpp
class Solution {
    int atMost(vector<int>& nums,int bound){
        int result=0,length=0;
        for(int value:nums){
            length=value<=bound?length+1:0;
            result+=length;
        }
        return result;
    }
    public:int numSubarrayBoundedMax(vector<int>& nums,int left,int right){
        return atMost(nums,right)-atMost(nums,left-1);
    }
};

File: content/problems/0795-number-of-subarrays-with-bounded-maximum/solutions/difference-of-at-most-counts/Solution.kt
class Solution {
    fun numSubarrayBoundedMax(nums:IntArray,left:Int,right:Int):Int=atMost(nums,right)-atMost(nums,left-1);
    private fun atMost(nums:IntArray,bound:Int):Int{
        var result=0;
        var length=0;
        for(value in nums){
            length=if(value<=bound)length+1 else 0;
            result+=length
        };
        return result
    }
}
